src/crimson/os/seastore/transaction.h

```diff
@@ -143,8 +143,16 @@ class Transaction {
   void add_to_read_set(CachedExtentRef ref) {
     if (is_weak()) return;
 
+    assert(ref->is_valid());
+
+    auto it = ref->transactions.lower_bound(
+      this, read_set_item_t<Transaction>::trans_cmp_t());
+    if (it != ref->transactions.end() && it->t == this) return;
```

Comment:
Because we now try to get extents following lba pointers and don't do `Transaction::get_extent()` most of the time, so we can't avoid repeatedly adding an extent to transactions' read_set

Reply:
Looks reasonable to me.